api/src/main/java/io/minio/MinioClient.java

```
@@ -1509,6 +1535,107 @@ public void getObject(String bucketName, String objectName, String fileName)
+   *          Object name in the bucket.
+   * @param encryptionMaterials
+   *          Encryption metadata details and operations.
+   *
+   * @return {@link InputStream} containing the object data.
+   *
+   * @throws InvalidBucketNameException
+   *           upon invalid bucket name is given
+   * @throws NoResponseException
+   *           upon no response from server
+   * @throws IOException
+   *           upon connection error
+   * @throws XmlPullParserException
+   *           upon parsing response xml
+   * @throws ErrorResponseException
+   *           upon unsuccessful execution
+   * @throws InternalException
+   *           upon internal library error
+   * @throws InvalidEncryptionMetadataException
+   *           upon encryption key/iv error
+   * @throws BadPaddingException
+   *           upon wrong padding
+   * @throws IllegalBlockSizeException
+   *           upon incorrect block size
+   * @throws NoSuchPaddingException
+   *           upon incorrect padding
+   * @throws InvalidAlgorithmParameterException
+   *           upon incorrect algorithm
+   */
+  public CipherInputStream getObject(String bucketName, String objectName, EncryptionMaterials encryptionMaterials)
+      throws InvalidBucketNameException, NoSuchAlgorithmException, InsufficientDataException, IOException,
+      InvalidKeyException, NoResponseException, XmlPullParserException, ErrorResponseException, InternalException,
+      InvalidArgumentException, InvalidEncryptionMetadataException, NoSuchPaddingException, IllegalBlockSizeException,
+      BadPaddingException, InvalidAlgorithmParameterException {
+
+    if (encryptionMaterials == null) {
+      throw new InvalidArgumentException("empty encryption properties not allowed");
+    }
+
+    // get encrypted object metadata and verify
+    ObjectStat stat = statObject(bucketName, objectName);
+
+    // check if encryption metadata is not present
+    if ((stat.contentKey() == null) || (stat.encryptionIv() == null)) {
```

I dont think falling back to regular `getObject` is a good option here. If an application needs plain object it should call corresponding `getObject` API. 

If we fall back to default `getObject` in cases where `encryptionMaterials` is null due to some error, we'll be silently discarding the error and returning wrong object. That is not correct.